src/tenant/tests/test_admin.py

```diff
@@ -93,19 +93,114 @@ class PublicTenantTestAdminPublic(TenantTestCase):
 
     def setUp(self):
         # create the public schema
-        self.public_tenant = Tenant(
-            schema_name='public',
-            name='public'
-        )
-
+        self.public_tenant = Tenant(schema_name="public", name="public")
+        self.tenant_model_admin = TenantAdmin(model=Tenant, admin_site=AdminSite())
         with tenant_context(self.public_tenant):
-            # Hack to create the public tenant without trigerring the signals
```

Comment:
The old comment didn't mention why this setup needed to prevent triggering signals.  Is there a reason you had to change it?

Reply:
Since "setUp" method run before each test, avoiding trigerring django signals (post_save and pre_save) can save us a lot of time.
 
The only exception is an "extra" tenant object, I need it to be a kind of "real" deck like one created by superuser in django admin, with all related objects created (SiteConfig, Domain and whatever else required now or in the future).

Related commit https://github.com/bytedeck/bytedeck/pull/1457/commits/304df60702d8dceaefbea7127e3a88b9da8e90df